spring-batch-s3/README.adoc

```diff
@@ -0,0 +1,157 @@
+
+* `S3StringSerializer`: take a `String` as input and writes it to S3 as a UTF-8 encoded byte array. The write functions add a line termination character at the end of each string.
+* `S3StringDeserializer`: takes a UTF-8 encoded byte array from S3 and converts it to a `String`. The implementation of this deserializer is *stateful* because lines may arrive in different chunks.
+
+More details in the JavaDocs of the classes.
+
+=== Configuration of the `S3Client`
+
+To use the `S3ItemReader` and `S3ItemWriter`, you need to configure the AWS S3 client by providing a `S3Client`. Checkout the https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/http-configuration-apache.html[AWS SDK for Java] documentation for more details on how to configure this.
+
+=== Configure `S3Serializer`
+
+`S3StringSerializer` is a simple implementation of `S3Serializer` that takes a `String` as input and writes it to S3 as a UTF-8 encoded byte array. You are encouraged to implement your own serializer if you need to handle different data types or formats.
+
+==== Java Config
+
+[source,java]
+----
+@Bean
+S3Serializer<String> s3Serializer() {
+    return new S3StringSerializer();
+}
+----
+
+=== Configure `S3Deserializer`
+
+Similarly, `S3StringDeserializer` is a simple implementation of `S3Deserializer` that takes a UTF-8 encoded byte array from S3 and converts it to a `String`. You can implement your own deserializer if you need to handle different data types or formats.
+
+In case you don't want to implement your serializer checkout the "Alternatives readers" section below.
+
+==== Java Config
+
+[source,java]
+----
+@Bean
+S3Deserializer<String> s3Deserializer() {
+    return new S3StringDeserializer();
+}
+----
+
+== Configuration of `S3ItemReader`
+
+Given the `S3Client` and `S3Deserializer` beans, you can now configure the `S3ItemReader`.
+
+=== Java Config
+
+To configure the `S3ItemReader`, you need to set up the AWS S3 client and specify the bucket and object key from which you want to read data.
+[source,java]
+----
+@Bean
+ItemReader<String> downloadItemReader() throws Exception {
+    return  new S3ItemReader.Builder<String>()
+            .s3Client(s3Client())
+            .bucketName("bucket_name")
+            .objectKey("object_key")
+            .deserializer(s3Deserializer())
+            .bufferSize((int) DataSize.ofMegabytes(1).toBytes()) // Default 128 Bytes
+            .build();
+}
+----
+
+There is also an additional option to set the `bufferSize` which is the size of the buffer used to read data from S3. The default value is 128 bytes, but you can increase it to improve memory consumption The bast value for this parameter is the average length of the lines in your file.
+
+=== Alternative reader
+
+Instead `S3ItemReader` you can also use  `FlatFileItemReader` with `InputStreamResources` to read files from S3 as well.
+To do so this package exposes a `S3InputStreamResource` that can be used for that purpose. Below an example:
```

Comment:
I think this is a typo, there is no `S3InputStreamResource` in this module, I guess you mean `S3InputStream`

Reply:
Correct, fixed it in next commit